packages/plugin/package.json

```diff
@@ -39,9 +39,15 @@
     "typecheck": "tsc --noEmit"
   },
   "peerDependencies": {
+    "@apollo/federation-internals": "^2",
     "eslint": ">=8.44.0",
     "graphql": "^16"
   },
+  "peerDependenciesMeta": {
+    "@apollo/federation-internals": {
+      "optional": true
+    }
+  },
   "dependencies": {
     "@graphql-tools/code-file-loader": "^8.0.0",
     "@graphql-tools/graphql-tag-pluck": "^8.0.0",
```

Comment:
maybe we should add `@graphql-tools/graphql-tag-pluck` also to `peerDependencies` in v4, since it's a big dependency that installs a lot of things (not everybody lint code files)